templates/default/ssm/write_aws_credentials.sh.erb

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# This script writes AWS credentials for the cluster admin user.
+# This script is meant to be called by SSM.
+#
+# Usage: $SCRIPT_NAME [AWS_IAM_ASSUMED_ROLE_ARN] [AWS_ACCESS_KEY_ID] [AWS_SECRET_ACCESS_KEY] [AWS_SESSION_TOKEN]
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+SCRIPT_NAME="$(basename "${BASH_SOURCE[0]}")"
+
+CLUSTER_ADMIN_USER=<%= node['cluster']['cluster_admin_user'] %>
+
+AWS_IAM_ASSUMED_ROLE_ARN=$1
+AWS_ACCESS_KEY_ID=$2
+AWS_SECRET_ACCESS_KEY=$3
+AWS_SESSION_TOKEN=$4
+
+function error () {
+  >&2 echo "ERROR: $1"
+  exit 1
+}
+
+echo "Running script: $SCRIPT_DIR/$SCRIPT_NAME"
+echo "Running as user: $(whoami)"
+echo "Running from folder: $(pwd)"
+echo "Running for cluster admin user: $CLUSTER_ADMIN_USER"
+
+[[ -n $AWS_IAM_ASSUMED_ROLE_ARN ]] || error "Required argument missing: AWS_IAM_ASSUMED_ROLE_ARN"
+[[ -n $AWS_ACCESS_KEY_ID ]] || error "Required argument missing: AWS_ACCESS_KEY_ID"
+[[ -n $AWS_SECRET_ACCESS_KEY ]] || error "Required argument missing: AWS_SECRET_ACCESS_KEY"
+[[ -n $AWS_SESSION_TOKEN ]] || error "Required argument missing: AWS_SESSION_TOKEN"
+
+[[ -n $(id -u $CLUSTER_ADMIN_USER 2>/dev/null) ]] || error "User does not exist: $CLUSTER_ADMIN_USER"
+
+CLUSTER_ADMIN_USER_HOME=$(sudo -u $CLUSTER_ADMIN_USER env | grep HOME | cut -d = -f 2)
```

Comment:
would it make sense to `grep -E` for `^HOME=` to guard against the case where default global envs containing `HOME` have been configured on a custom AMI?

Reply:
Sure. In case the envar HOME is not set, the script should fail rather than creating the dir tree because the AWS CLI would fail if it is not set.

Reply:
Sure, thanks. The script should fail when the envar `HOME` is not set because the AWS CLI would fail if it is not set.